sw/device/tests/sim_dv/sysrst_ctrl_in_irq_test.c

```diff
@@ -0,0 +1,163 @@
+    kDifSysrstCtrlPinKey0In,           kDifSysrstCtrlPinKey1In,
+    kDifSysrstCtrlPinKey2In,           kDifSysrstCtrlPinPowerButtonIn,
+    kDifSysrstCtrlPinAcPowerPresentIn, kDifSysrstCtrlPinLidOpenIn,
+    kDifSysrstCtrlPinEcResetInOut,     kDifSysrstCtrlPinFlashWriteProtectInOut,
+};
+
+dif_sysrst_ctrl_input_change_config_t config = {
+  .input_changes = kDifSysrstCtrlInputKey0L2H,
+  .debounce_time_threshold = 1,  // 5us
+};
+
+/**
+ * External interrupt handler.
+ */
+void ottf_external_isr(void) {
+  LOG_INFO("DEBUG: enter ottf external");
+
+  dif_rv_plic_irq_id_t irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+  LOG_INFO("DEBUG: irq claimed");
+
+  peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralSysrstCtrlAon) {
+    LOG_INFO("DEBUG: true peripheral = %d", peripheral);
+    irq = (dif_sysrst_ctrl_irq_t)(
+        irq_id -
+        (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdSysrstCtrlAonEventDetected);
+
+    if (irq_id == kTopEarlgreyPlicIrqIdSysrstCtrlAonEventDetected) {
+      LOG_INFO("DEBUG: true irq_id = %d", irq_id);
+    } else {
+      LOG_INFO("DEBUG: false irq_id = %d", irq_id);
+    }
+
+    CHECK_DIF_OK(dif_sysrst_ctrl_irq_acknowledge(&sysrst_ctrl, irq));
+    LOG_INFO("DEBUG: irq acked");
+  } else {
+    LOG_INFO("DEBUG: false peripheral = %d", peripheral);
+  }
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC.
+  // register
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kPlicTarget, irq_id));
+  LOG_INFO("DEBUG: irq completed");
+}
+
+bool test_main(void) {
+  LOG_INFO("DEBUG: Enter SW main");
+
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+  LOG_INFO("DEBUG: irq enabled");
+
+  // Initialize the PLIC.
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(plic_base_addr, &plic));
+  LOG_INFO("DEBUG: PLIC inited");
+
+  // Enable all the SYSRST CTRL interrupts on PLIC
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdSysrstCtrlAonEventDetected,
+      kTopEarlgreyPlicIrqIdSysrstCtrlAonEventDetected);
+  LOG_INFO("DEBUG: irq range enabled");
+
+  // Initialize sysrst ctrl
+  CHECK_DIF_OK(dif_sysrst_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR),
+      &sysrst_ctrl));
+
+  // Set input pins
+  dif_pinmux_t pinmux;
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  for (int i = 0; i < kOutputNunMioPads; ++i) {
+    CHECK_DIF_OK(
+        dif_pinmux_input_select(&pinmux, kPeripheralInputs[i], kInputPads[i]));
+  }
+
+  // Configure sysrst ctrl to detect input change and enable irq on sysrst ctrl
+  CHECK_DIF_OK(dif_sysrst_ctrl_input_change_detect_configure(&sysrst_ctrl, config));
```

Comment:
I would wrap from line 135 to 159 inside a fuction so that it can be reused to test other pins.